.github/workflows/ci.yml

```diff
@@ -0,0 +1,96 @@
+      - name: list module files to lint
+        run: |
+          [ -d "modules/azure" ] && ls -li "modules/azure"
+      - name: lint modules/azure folder
+        id: azure_module_lint
+        uses: golangci/golangci-lint-action@v2.2.0
+        with:
+          version: v1.30
+          working-directory: modules/azure
+      - name: list test files to lint
+        run: |
+          [ -d "test/azure" ] && ls -li "test/azure"
+      - name: lint test/azure folder
+        id: azure_test_lint
+        uses: golangci/golangci-lint-action@v2.2.0
+        with:
+          version: v1.30
+          working-directory: test/azure
+      - name: login to azure cli
+        uses: azure/login@v1.1
+        with:
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+      - name: run go test for azure
+        id: azure_test
+        env:
+          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+        run: |
+          az account show
+          cd test
+          TEST_VERSION=`date +"%Y%m%d%H%M"`
+          export ARM_SUBSCRIPTION_ID=`az account show --query "id" --output tsv`
+
+          APP_ID=`echo $AZURE_CREDENTIALS | jq -r -c ".clientId"`
+
+          APP_PASSWORD=`echo $AZURE_CREDENTIALS | jq -r -c ".clientSecret"`
+
+          TENANT_ID=`echo $AZURE_CREDENTIALS | jq -r -c ".tenantId"`
+
+          export ARM_CLIENT_ID="$APP_ID"
+          export ARM_CLIENT_SECRET="$APP_PASSWORD"
+          export ARM_SUBSCRIPTION_ID="$ARM_SUBSCRIPTION_ID"
+          export ARM_TENANT_ID="$TENANT_ID"
+
+          rm -rf ssh_key*
+          ssh-keygen -m PEM -t rsa -b 4096 -f ./ssh_key -q -N ""
+          export TF_VAR_ssh_public_key="$PWD/ssh_key.pub"
+          export TF_VAR_client_id="$APP_ID"
+          export TF_VAR_client_secret="$APP_PASSWORD"
+
+          go test ./azure/* -timeout 90m
+      - name: report back the result
+        if: always()
+        env:
+          CURRENT_REPOSITORY: ${{ github.repository }}
+          RUN_ID: ${{ github.run_id }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          TARGET_REPOSITORY: ${{ github.event.inputs.target_repository }}
+          TARGET_PR: ${{ github.event.inputs.target_pr }}
+          TEST_RESULT: ${{ steps.azure_test.conclusion }}
+          TEST_LINT_RESULT: ${{ steps.azure_test_lint.conclusion }}
+          MODULE_LINT_RESULT: ${{ steps.azure_module_lint.conclusion }}
+        run: |
+          if [ "$TEST_RESULT" == "success" ] && [ "$TEST_LINT_RESULT" == "success" ] && [ "$MODULE_LINT_RESULT" == "success" ]; then
+            BODY_PAYLOAD="[Microsoft CI Bot] TL;DR; success :thumbsup:\n\nYou can check the status of the CI Pipeline logs here ; https://github.com/${CURRENT_REPOSITORY}/actions/runs/$RUN_ID"
+            curl --request POST --header "Authorization: token ${GITHUB_TOKEN}" --header "Accept: application/vnd.github.v3+json" https://api.github.com/repos/${TARGET_REPOSITORY}/issues/${TARGET_PR}/comments --data "{\"body\":\"${BODY_PAYLOAD}\"}"
```

Comment:
So these `curl` calls would update the test status on the official Terratest repo?

Reply:
correct, @polatengin can confirm but I am pretty sure that is how it works.

Reply:
at the end of the run, CI pipeline creates a comment on the PR, saying either "success" or "failure" and the link to the run logs.

curl call creates that comment.

I just noticed, I should add some comments to make it more clear.

Reply:
added couple of comments to make the intent clear